android-client-sdk/src/main/java/com/devcycle/sdk/android/openfeature/DevCycleEventMapper.kt

```diff
@@ -0,0 +1,58 @@
+package com.devcycle.sdk.android.openfeature
+
+import com.devcycle.sdk.android.model.DevCycleEvent
+import com.devcycle.sdk.android.util.DevCycleLogger
+import dev.openfeature.sdk.TrackingEventDetails
+import dev.openfeature.sdk.Value
+import java.math.BigDecimal
+
+object DevCycleEventMapper {
+    
+    /**
+     * Converts OpenFeature tracking event details to a DevCycle event
+     * 
+     * @param trackingEventName The name/type of the event to track
+     * @param details Optional tracking event details containing value and metadata
+     * @return DevCycleEvent ready to be tracked
+     */
+    fun openFeatureEventToDevCycleEvent(
+        trackingEventName: String,
+        details: TrackingEventDetails?
+    ): DevCycleEvent {
+        val builder = DevCycleEvent.builder().withType(trackingEventName)
+        
+        // Process numeric value if provided
+        details?.value?.let { value ->
+            builder.withValue(BigDecimal.valueOf(value.toDouble()))
+        }
+        
+        // Process metadata from structure if provided - unwrap Value objects to raw values
+        details?.structure?.asMap()?.let { detailData ->
+            val metadata = mutableMapOf<String, Any>()
+            detailData.forEach { (key, value) ->
+                val unwrappedValue = unwrapValue(value)
+                if (unwrappedValue != null) {
+                    metadata[key] = unwrappedValue
+                }
+            }
+            if (metadata.isNotEmpty()) {
+                builder.withMetaData(metadata)
+            }
+        }
+        
+        return builder.build()
+    }
+    
+    /**
+     * Unwraps OpenFeature Value objects to raw Java types
+     */
+    private fun unwrapValue(value: Any): Any? {
```

Comment:
we accept metadata as <string,Any> , this does not allow for any nested values

Reply:
I am also unsure if nested valued are allowed to be part of the Immutable Structure as that is the type of OpenFeature event

Reply:
nvm, i see that the Immutable Structure does allow maps in there,  but this will remove those values